bootstrap/interfaces/secret.go

```diff
@@ -28,4 +28,10 @@ type SecretProvider interface {
 
 	// HasSecret returns true if the service's SecretStore contains a secret at the specified path.
 	HasSecret(path string) (bool, error)
+
+	// RegisteredSecretUpdatedCallback registers a callback for a secret.
+	RegisteredSecretUpdatedCallback(path string, callback func(path string))
```

Comment:
Make this function return `(func(), error)`.  When called, the returned func() will remove the registered callback from the map (only if path and callback match).